doc/experiment.rst

```diff
@@ -1,327 +1,423 @@
-
-  # setup ensemble parameter space
-  learning_rate = list(np.linspace(.01, .5))
-  train_params = {"LR": learning_rate}
-
-  # define resources for all ensemble members
-  sbatch = exp.create_batch_settings(nodes=4,
-                                    time="01:00:00",
-                                    account="12345-Cray",
-                                    queue="gpu")
-
-  # define how each member should run
-  srun = exp.create_run_settings(exe="python",
-                                exe_args="./train-model.py")
-  srun.set_nodes(1)
-  srun.set_tasks(24)
-
-  ensemble = exp.create_ensemble("Training-Ensemble",
-                                params=train_params,
-                                params_as_args=["LR"],
-                                batch_settings=sbatch,
-                                run_settings=srun,
-                                perm_strategy="random",
-                                n_models=4)
-
-  # Enable key prefixing -- note that this should be done
-  # before starting the experiment
-  ensemble.enable_key_prefixing()
-
-  exp.start(ensemble, summary=True)
-
-
-Further Information
--------------------
-
-For more informtion about Ensembles, please refer to the :ref:`Ensemble API documentation <ensemble_api>`.
\ No newline at end of file
+Create a Ensemble
+^^^^^^^^^^^^^^^^^
+An ensemble is created through the function: ``Experiment.create_ensemble()``. The function requires
+one of the subsequent sets of arguments upon initialization:
+
+Case 1 : ``RunSettings`` and `params` or `replicas`
+    If it only passed RunSettings, Ensemble, objects will
+    require either a replicas argument or a params argument to
+    expand parameters into ``Model`` instances.
+    At launch, the Ensemble will look for interactive allocations to launch models in.
+
+Case 2 : ``BatchSettings``
+    If it passed BatchSettings without other arguments,
+    an empty Ensemble will be created that ``Model`` objects
+    can be added to manually. All ``Model`` objects added to
+    the Ensemble will be launched in a single batch.
+
+Case 3 : ``BatchSettings``, `run_settings`, and `params`
+    If it passed BatchSettings and RunSettings, the BatchSettings
+    will determine the allocation settings for the entire batch,
+    and the RunSettings will determine how each individual Model
+    instance is executed within that batch.
+
+Case 4 : ``BatchSettings``, ``RunSettings``, and `replicas`
+    If each of multiple ensemble members attempt to use the
+    same code to access their respective models in the Orchestrator,
+    the keys by which they do this will overlap and they can end up
+    accessing each others’ data inadvertently. To prevent
+    this situation, the SmartSim Entity object supports
+    key prefixing, which automatically prepends the name
+    of the model to the keys by which it is accessed. With
+    this enabled, key overlapping is no longer an issue and
+    ensemble members can use the same code.
```

Comment:
Give an example for what this looks like:
Assume you have two models in the Ensemble object, named `bar_0` and `bar_1`. In the application code you'll have code like this "client.put_tensor("foo"). With ensemble key prefixing turned on, the actual key that will end up in the database is `bar_0_foo` and `bar_1_foo`.